AsyncTwoPhaseIndexerTests failure

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/3534/testReport/junit/org.elasticsearch.xpack.core.indexing/AsyncTwoPhaseIndexerTests/testStateMachine/

```
Caused by: java.lang.AssertionError: 
Expected: <3>
     but: was <5>
	at __randomizedtesting.SeedInfo.seed([5119E6D95F3BD4B0]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexerTests$MockIndexer.doSaveState(AsyncTwoPhaseIndexerTests.java:113)
```

Does not reproduce
```
/gradlew :x-pack:plugin:core:test --tests "org.elasticsearch.xpack.core.indexing.AsyncTwoPhaseIndexerTests.testStateMachine" -Dtests.seed=5119E6D95F3BD4B0 -Dtests.security.manager=true -Dtests.locale=xog -Dtests.timezone=CTT -Dcompiler.java=12 -Druntime.java=11
```

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexing/AsyncTwoPhaseIndexerTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexing/AsyncTwoPhaseIndexerTests.java
@@ -64,7 +64,7 @@ public class AsyncTwoPhaseIndexerTests extends ESTestCase {
 
         @Override
         protected IterationResult<Integer> doProcess(SearchResponse searchResponse) {
-            awaitForLatch();
+            assertFalse("should not be called as stoppedBeforeFinished is false", stoppedBeforeFinished);
             assertThat(step, equalTo(3));
             ++step;
             return new IterationResult<>(Collections.emptyList(), 3, true);
@@ -99,6 +99,9 @@ public class AsyncTwoPhaseIndexerTests extends ESTestCase {
             final SearchResponseSections sections = new SearchResponseSections(
                 new SearchHits(new SearchHit[0], new TotalHits(0, TotalHits.Relation.EQUAL_TO), 0), null,
                 null, false, null, null, 1);
+
+            // block till latch has been counted down, simulating network latency
+            awaitForLatch();
             nextPhase.onResponse(new SearchResponse(sections, null, 1, 1, 0, 0, ShardSearchFailure.EMPTY_ARRAY, null));
         }
 
@@ -222,7 +225,6 @@ public class AsyncTwoPhaseIndexerTests extends ESTestCase {
         }
     }
 
-    @AwaitsFix( bugUrl = "https://github.com/elastic/elasticsearch/issues/42084")
     public void testStateMachine() throws Exception {
         AtomicReference<IndexerState> state = new AtomicReference<>(IndexerState.STOPPED);
         final ExecutorService executor = Executors.newFixedThreadPool(1);
@@ -265,7 +267,6 @@ public class AsyncTwoPhaseIndexerTests extends ESTestCase {
         }
     }
 
-    @AwaitsFix( bugUrl = "https://github.com/elastic/elasticsearch/issues/42084")
     public void testStop_AfterIndexerIsFinished() throws InterruptedException {
         AtomicReference<IndexerState> state = new AtomicReference<>(IndexerState.STOPPED);
         final ExecutorService executor = Executors.newFixedThreadPool(1);
@@ -285,7 +286,6 @@ public class AsyncTwoPhaseIndexerTests extends ESTestCase {
         }
     }
 
-    @AwaitsFix( bugUrl = "https://github.com/elastic/elasticsearch/issues/42084")
     public void testStop_WhileIndexing() throws InterruptedException {
         AtomicReference<IndexerState> state = new AtomicReference<>(IndexerState.STOPPED);
         final ExecutorService executor = Executors.newFixedThreadPool(1);

